### PR TITLE
Restructure source

### PR DESCRIPTION
**Pull request by tranhuyducseven**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,11 @@
...
+node_modules
+
+**.log
+
+# local env files
+*.env*
+
+# cache
+.eslintcache
+**.vscode
```
Comment (vinhtrand8) [inline .gitignore:11]: remove `**.vscode`

File: .husky/.gitignore
Comment (vinhtrand8) [inline .husky/.gitignore:1]: delete this file

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo "Running husky pre-commit hook"
+bun lint && bun lintc
```
Comment (vinhtrand8) [inline .husky/pre-commit:5]: add new line

File: contracts/.gitignore [hunk 3]
```diff
@@ -18,3 +18,4 @@ yarn.lock
 !broadcast
 broadcast/*
 broadcast/*/31337/
+.vscode
```
Comment (vinhtrand8) [inline contracts/.gitignore:21]: remove this line

File: contracts/.vscode/settings.json
Comment (vinhtrand8) [inline contracts/.vscode/settings.json:1]: recover it

File: contracts/package.json [hunk 4]
```diff
@@ -13,8 +13,7 @@
     "@openzeppelin/upgrades-core": "^1.34.1",
     "chalk": "^5.3.0",
     "ethers": "^6.13.1",
-    "openzeppelin-foundry-upgrades": "github:OpenZeppelin/openzeppelin-foundry-upgrades",
-    "solidity-stringutils": "github:Arachnid/solidity-stringutils"
```
Comment (vinhtrand8) [inline contracts/package.json:17]: use `solidity-stringutils` for upgradable contract

Comment (tranhuyducseven) [reply contracts/package.json]: image

**Review by vinhtrand8 — APPROVED** [review]

**Pull request merged**